### PR TITLE
Backend setup

### PR DESCRIPTION
**Pull request by JesseNaidoo**

### CODE REVIEW

File: api/utils/http-status-codes.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const OK = 200;
```
Comment (ryanbasiltrickett) [inline api/utils/http-status-codes.js]: Why not make this structure an enum?

File: terraform/api.tf [hunk 2]
```diff
@@ -110,4 +114,29 @@ resource "aws_elastic_beanstalk_environment" "beanstalk_env" {
...
+  }
+  setting {
+    namespace = "aws:elasticbeanstalk:application:environment"
+    name      = "BD_PORT"
+    value     = module.rds.db_instance_port
+  }
+  setting {
+    namespace = "aws:elasticbeanstalk:application:environment"
+    name      = "DB_NAME"
+    value     = "spyderdb"
```
Comment (ryanbasiltrickett) [inline terraform/api.tf]: You should be able to say module.rds.db_name or something along those lines

File: api/controllers/account.controller.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
+import {HttpStatusCodes, DbUtils} from '../utils/index.js';
+
+export const createAccount = async (req, res) => {
+    try {
+        const {uid, username} = req.body;
+        if (!uid || !username) {
+            res.status(HttpStatusCodes.BadRequest).send({message: "Invalid payload"});
+        } else {
+            await DbUtils.spyderdb.oneOrNone('SELECT * FROM account WHERE uid = $1 OR username = $2', [uid, username])
```
Comment (ryanbasiltrickett) [inline api/controllers/account.controller.js:9]: Database munipulation should only be done in repositories

File: api/controllers/account.controller.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+                .then(async data => {
```
Comment (ryanbasiltrickett) [inline api/controllers/account.controller.js:10]: Logic should only be done in services

File: api/.idea/.gitignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Default ignored files
```
Comment (ryanbasiltrickett) [inline api/.idea/.gitignore]: .idea directories should be ignored and not in git

**Review by ryanbasiltrickett — CHANGES_REQUESTED** [review]

File: api/controllers/account.controller.js (hunk 3 above)
Comment (JesseNaidoo) [reply api/controllers/account.controller.js:9]: So I do agree with this, just that due to time constraints and the fact that Rudolph adn Thabang will not look at the backend. I think we should leave it this.

File: api/controllers/account.controller.js (hunk 4 above)
Comment (JesseNaidoo) [reply api/controllers/account.controller.js:10]: Same goes here, it does make sense to have services and repositories. But for a small project like this I think it is a bit of an overkill.

**Review by ryanbasiltrickett — APPROVED** [review]
LGTM

**Pull request merged**